tutorials/eth-api/thirdweb.md

```diff
@@ -0,0 +1,285 @@
+
+thirdweb offers a small number of chains from `@thirdweb/chains` and does not include Moonbeam networks in that list, so you'll need to specify the network details including chain ID and RPC URL. You can create a custom chain with [`defineChain`](https://portal.thirdweb.com/references/typescript/v5/defineChain){target=\_blank} as follows:
+
+=== "Moonbeam"
+
+    ```typescript
+    import { defineChain } from 'thirdweb';
+    const moonbase = defineChain({
+    id: {{ networks.moonbeam.chain_id }},
+    rpc: '{{ networks.moonbeam.public_rpc_url }}',
+    })
+    ```
+
+=== "Moonriver"
+
+    ```typescript
+    import { defineChain } from 'thirdweb';
+    const moonbase = defineChain({
+    id: {{ networks.moonriver.chain_id }},
+    rpc: '{{ networks.moonriver.public_rpc_url }}',
+    })
+    ```
+
+=== "Moonbase Alpha"
+
+    ```typescript
+    import { defineChain } from 'thirdweb';
+    const moonbase = defineChain({
+    id: {{ networks.moonbase.chain_id }},
+    rpc: '{{ networks.moonbase.rpc_url }}',
+    })
+    ```
+
+The NFT marketplace template includes a `chains.ts` file under `src/consts/chains.ts`. To add support for Moonbase Alpha to the dApp, add the following lines: 
+
+```typescript title="chains.ts"
+export const moonbase = defineChain({
+  id: 1287,
+  rpc: 'https://rpc.api.moonbase.moonbeam.network',
+});
+```
+
+We don't need to add any import statements because `defineChain` is already imported by default as part of the template. Feel free to add additional chains if you'd like to add support for Moonbeam, Moonriver, or other networks. The full file can be viewed below:
+
+
+??? code "View chains.ts"
+    ```typescript
+    --8<-- 'code/tutorials/eth-api/thirdweb/chains.ts'
+    ```
+
+
+## Deploy ERC-721 NFT Contract {: #deploy-erc-721-nft-contract }
+
+Of course, we'll need to have an NFT contract to showcase as part of the marketplace. You can use an existing NFT contract, but for demo purposes we'll walk through the steps of deploying a new ERC-721 contract with thirdweb. 
+
+Head to [thirdweb explore](https://thirdweb.com/explore){target=\_blank} and choose the `OpenEditionERC721` NFT standard. You can also access the [NFT contract directly](https://thirdweb.com/thirdweb.eth/OpenEditionERC721){target=\_blank}. Press **Deploy Now**, then take the following steps:
+
+1. Add a name for your NFT
+2. Optionally add a token symbol
+3. Upload the image for your NFT. This will be uploaded to IPFS
+4. Review the royalty information. By default, this is set to the address of your currently connected wallet and the royalty is set by default to 0%
+5. Review the address to receive the proceeds of initial NFT sales. By default, this is set to the address of your currently connected wallet
+6. Select your Network as **Moonbase Alpha**
+7. Press **Deploy Now**
+
+![Configure ERC-721](/images/tutorials/eth-api/thirdweb/thirdweb-2.webp "Configuring an ERC-721 contract on thirdweb")
```

Comment:
we have a new way of doing images. this is fine for now, but just wanted to give you a heads up. details are in the wiki under docs standards